jax/_src/sourcemap.py

```diff
@@ -0,0 +1,182 @@
+from dataclasses import dataclass
+from typing import Iterable
+
+### An implementation of sourcemaps following TC39 (see https://tc39.es/source-map/)
+
+# Each segment is made up of 1, 4 or 5 variable-length fields
+# (see https://tc39.es/source-map/#mappings-structure).
+Segment = tuple[int] | tuple[int, int, int, int] | tuple[int, int, int, int, int]
```

Comment:
We still have  CI build using Python 3.9. It might suffice to add `from __future__ import annotations`, but I think direct aliases like this have to be spelled with `Union` syntax not `|`.

Reply:
Just saw this in CI as well, I went with `Union`.